src/screening/bloodSugar/BloodSugarTestForm.jsx

```diff
@@ -5,39 +5,36 @@ import { Grid, Row, FormGroup, ControlLabel, Col } from 'react-bootstrap';
 import { ENCOUNTER_TYPES, CONCEPTS, FORM_ANSWERS} from "../../constants";
 import '../../assets/css/tabs.css';
 import ScreeningForm from "../ScreeningForm";
-import { formValueSelector } from "redux-form";
+import {change, formValueSelector, untouch} from "redux-form";
 import { noPaddingLeftAndRight, flexBaseline } from "../../pwaStyles";
 
 
 class BloodSugarTestForm extends React.PureComponent {
 
+  componentDidUpdate(prevProps) {
+    if (typeof this.props.fsTestType.value !== 'undefined' &&
+      typeof prevProps.fsTestType.value !== 'undefined' &&
+      this.props.fsTestType.value !== prevProps.fsTestType.value) {
+      this.clearField(this.props.bloodGlucose.fieldName);
+    }
+  }
+
+  clearField(field) {
+    this.props.dispatch(change(this.props.formInstanceId, field, null));
+    this.props.dispatch(untouch(this.props.formInstanceId, field));
+  }
+
   render() {
 
     const {
       fsTestType
     } = this.props;
 
+    let glucoseConcept = Object.assign({}, CONCEPTS.BLOOD_GLUCOSE);
 
-    let fastingConcept = Object.assign({}, CONCEPTS.RANDOM_BLOOD_GLUCOSE, { hiNormal: 126 });
-
-
-    const randomObs = (
-      <Obs
-        concept={ CONCEPTS.RANDOM_BLOOD_GLUCOSE }
-        placeholder="value"
-        path="random-glucose"
-        required={ true }
-      />
-    );
-
-    const fastingObs = (
-      <Obs
-        concept={ fastingConcept }
-        placeholder="value"
-        path="fasting-glucose"
-        required={ true }
-      />
-    );
+    if ((typeof fsTestType.value !== 'undefined') && (fsTestType.value === CONCEPTS.FASTING_BLOOD_SUGAR_TEST.uuid)) {
+      glucoseConcept.hiNormal = 126;
+    }
```

Comment:
I changed this so that there's just a single glucoseConcept, that it clones from the store but then modifies the high normal if it's a fasting blood sugar.  But to get this to work I had to add something to React Components to re-initialize the validations when this changes:

https://github.com/openmrs/openmrs-react-components/commit/8cc6f112e1cca593b8f9e79414a06f378bd0528c